### ISSUE TITLE
Introduction of Interfaces quite fast

### ISSUE THREAD

**Issue by psteinb** [carpentries-incubator/julia-novice#49]
The following lines [06-control-flow.md#L44 ](https://github.com/carpentries-incubator/julia-novice/blob/249f6928ba6e1b0e70fb540840b101a36195d344/_episodes/06-control-flow.md?plain=1#L44)-[06-control-flow.md#L62 ](https://github.com/carpentries-incubator/julia-novice/blob/249f6928ba6e1b0e70fb540840b101a36195d344/_episodes/06-control-flow.md?plain=1#L62) introduce interfaces quite fast. 

I think this might put learners off, especially those that come with a background of scripting languages mostly. I personally would encourage to invest some text to introduce the notion of an interface more slowly. Maybe using a number (think interface `Real`) as an example or using `LinRange` as an implementation of `AbstractRange`.

**Comment by kescobo** [comment]
I agree. In fact, I think the notion of interfaces _might_ be outside the scope of a beginner course. I rarely use interfaces explicitly, and many interfaces in julia are poorly specified. 

That said, the array interface is one of the well-specified ones, and pretty fundamental to a lot of science-relevant code. Understanding it can be quite a boon, since many types implement `AbstractArray` or similar interfaces, and this can allow you to write faster and more generic code.

I definitely think that this section could use a bit more explication, and could be part of a broader re-write of this section (along the lines of #50

**Comment by BeastyBlacksmith** [comment]
@all_contributors add @kescobo for ideas
